src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java

```diff
@@ -0,0 +1,230 @@
+
+    /**
+     * Check if deliveryman is allowed to access warehouse.
+     * @param buildingWorker the building of the deliveryman.
+     * @return true if able to.
+     */
+    public boolean canAccessWareHouse(BuildingDeliveryman buildingWorker)
+    {
+        return registeredDeliverymen.contains(new Vec3d(buildingWorker.getID()));
+    }
+
+    /**
+     * Get the deliverymen connected with this building.
+     * @return the unmodifiable List of positions of them.
+     */
+    public List<Vec3d> getRegisteredDeliverymen()
+    {
+        return Collections.unmodifiableList(registeredDeliverymen);
+    }
+
+    @NotNull
+    @Override
+    public String getSchematicName()
+    {
+        return WAREHOUSE;
+    }
+
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return 5;
+    }
+
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.readFromNBT(compound);
+
+        registeredDeliverymen.clear();
+        final NBTTagList deliverymanTagList = compound.getTagList(TAG_DELIVERYMAN, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < deliverymanTagList.tagCount(); i++)
+        {
+            final BlockPos pos = NBTUtil.getPosFromTag(deliverymanTagList.getCompoundTagAt(i));
+            if(getColony() != null && getColony().getBuilding(pos) instanceof AbstractBuildingWorker && !registeredDeliverymen.contains(new Vec3d(pos)))
+            {
+                registeredDeliverymen.add(new Vec3d(pos));
+            }
+        }
+    }
+
+    @Override
+    public void writeToNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.writeToNBT(compound);
+        @NotNull final NBTTagList levelTagList = new NBTTagList();
+        for (@NotNull final Vec3d deliverymanBuilding : registeredDeliverymen)
+        {
+            levelTagList.appendTag(NBTUtil.createPosTag(new BlockPos(deliverymanBuilding)));
+        }
+        compound.setTag(TAG_DELIVERYMAN, levelTagList);
+    }
+
+    /**
+     * Returns the tile entity that belongs to the colony building.
+     *
+     * @return {@link TileEntityColonyBuilding} object of the building.
+     */
+    @Override
+    public TileEntityWareHouse getTileEntity()
+    {
+        final Colony colony = getColony();
+        if ((tileEntity == null || tileEntity.isInvalid()) && colony != null && colony.getWorld().getBlockState(this.getLocation()).getBlock() != null)
+        {
+            final TileEntity te = getColony().getWorld().getTileEntity(this.getLocation());
+            if (te instanceof TileEntityWareHouse)
+            {
+                tileEntity = (TileEntityWareHouse) te;
+                if (tileEntity.getBuilding() == null)
+                {
+                    tileEntity.setColony(colony);
+                    tileEntity.setBuilding(this);
+                }
+            }
+        }
+
+        return tileEntity;
+    }
+
+    @Override
+    public void serializeToView(@NotNull final ByteBuf buf)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method to simply inherit it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1185)